src/redux/book/booksSlice.js

```diff
@@ -1,43 +1,47 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
 
 const initialState = {
-  totalbooks: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  totalbooks: [],
 };
 
+export const fetchUsers = createAsyncThunk('user/fetchUsers', async () => {
+  try {
+    const response = await axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/O92kginp6T1sTL75SMAS/books');
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
+
+export const addUsers = createAsyncThunk('user/addUsers', async (data) => {
+  try {
+    const response = await axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/O92kginp6T1sTL75SMAS/books',
+      data);
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
+
+export const deleteUsers = createAsyncThunk('user/addUsers', async (id) => {
+  try {
+    const response = await axios.delete(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/O92kginp6T1sTL75SMAS/books/${id}`);
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
```

Comment:
- [ ] Well! It is important to ensure that deleting a book reflects in both the Redux store and the DOM to enhance the user experience. However, it appears that the DOM does not update when a book is deleted at the moment. To resolve this issue, kindly update your code so that deleting a book triggers the required updates in both the Redux store and the DOM.

_note: Currently your DOM and redux store are updating because of the continuous dispatching of `fetchUsers()`. As I mentioned earlier to resolve it, you need extra functionality to update your DOM and redux store upon deleting a book._ 

By implementing this change, you'll ensure that your application remains consistent and user-friendly when deleting books. Keep up the great work! 👍📚🔄